### ISSUE TITLE
In 8.0.0-preview.3.24105.21 project links to non-Aspire projects are now ignored for build

### ISSUE THREAD

**Issue by cisionmarkwalls** [dotnet/aspire#2234]
In 8.0.0-preview.3.24105.21 and VS Version 17.9.0 when I add a link to a project library for supporting services it is ignored by the build system.

Steps to Reproduce:

1. Create an Aspire app using the standard VS template
2. Create a support library project and add any class to it
3. Reference that project in the Aspire apphost
4. Attempt to use the namespace for that support class from the external library

The build process refuses to accept those references when building (saying the namespace does not exist) while this is set:
```
 <PropertyGroup>
   <OutputType>Exe</OutputType>
   <TargetFramework>net8.0</TargetFramework>
   <ImplicitUsings>enable</ImplicitUsings>
   <Nullable>enable</Nullable>
   **<IsAspireHost>true</IsAspireHost>**
 </PropertyGroup>
```
If you remove the IsAspireHost setting though it will see the reference/namespace and build successfully.

Why is this important?
* If you are using EF core and need to migrate the db created in the AppHost before the services come up you can't reference it.
* You can't create any supporting libraries for Aspire to handle your internal processes

**Comment by maranmaran** [comment]
> In 8.0.0-preview.3.24105.21 and VS Version 17.9.0 when I add a link to a project library for supporting services it is ignored by the build system.
> 
> Steps to Reproduce:
> 
> 1. Create an Aspire app using the standard VS template
> 2. Create a support library project and add any class to it
> 3. Reference that project in the Aspire apphost
> 4. Attempt to use the namespace for that support class from the external library
> 
> The build process refuses to accept those references when building (saying the namespace does not exist) while this is set:
> 
> ```
>  <PropertyGroup>
>    <OutputType>Exe</OutputType>
>    <TargetFramework>net8.0</TargetFramework>
>    <ImplicitUsings>enable</ImplicitUsings>
>    <Nullable>enable</Nullable>
>    **<IsAspireHost>true</IsAspireHost>**
>  </PropertyGroup>
> ```
> 
> If you remove the IsAspireHost setting though it will see the reference/namespace and build successfully.
> 
> Why is this important?
> 
> * If you are using EF core and need to migrate the db created in the AppHost before the services come up you can't reference it.
> * You can't create any supporting libraries for Aspire to handle your internal processes

So if I understand correctly you're referencing a library?

Can you try appending `IsAspireProjectResource="false" `?

` <ProjectReference Include="MyLib.csproj" IsAspireProjectResource="false" />`

**Comment by cisionmarkwalls** [comment]
That works! :) I should have thought of that. Thank you.
Is this going to be a long term way of dealing with that, or is this a short term fix?

**Comment by maranmaran** [comment]
> That works! :) I should have thought of that. Thank you. Is this going to be a long term way of dealing with that, or is this a short term fix?

I don't know I'm not a contributor

#1726 and https://github.com/dotnet/aspire/pull/1726#discussion_r1465440286
Maybe help here

**Comment by murh-lego** [comment]
> In 8.0.0-preview.3.24105.21 and VS Version 17.9.0 when I add a link to a project library for supporting services it is ignored by the build system.
> 
> Steps to Reproduce:
> 
> 1. Create an Aspire app using the standard VS template
> 2. Create a support library project and add any class to it
> 3. Reference that project in the Aspire apphost
> 4. Attempt to use the namespace for that support class from the external library
> 
> The build process refuses to accept those references when building (saying the namespace does not exist) while this is set:
> 
> ```
>  <PropertyGroup>
>    <OutputType>Exe</OutputType>
>    <TargetFramework>net8.0</TargetFramework>
>    <ImplicitUsings>enable</ImplicitUsings>
>    <Nullable>enable</Nullable>
>    **<IsAspireHost>true</IsAspireHost>**
>  </PropertyGroup>
> ```
> 
> If you remove the IsAspireHost setting though it will see the reference/namespace and build successfully.
> 
> Why is this important?
> 
> * If you are using EF core and need to migrate the db created in the AppHost before the services come up you can't reference it.
> * You can't create any supporting libraries for Aspire to handle your internal processes

Btw I on the other hand don't seem to can resolve project references. All `Microsoft.NET.Sdk.Web` projects are red to me on v1 with latest tooling VS 17.10 and 8.0.200

The `Microsoft.NET.Sdk` projects are fine

**Comment by jjhayter** [comment]
> > In 8.0.0-preview.3.24105.21 and VS Version 17.9.0 when I add a link to a project library for supporting services it is ignored by the build system.
> > Steps to Reproduce:
> > 
> > 1. Create an Aspire app using the standard VS template
> > 2. Create a support library project and add any class to it
> > 3. Reference that project in the Aspire apphost
> > 4. Attempt to use the namespace for that support class from the external library
> > 
> > The build process refuses to accept those references when building (saying the namespace does not exist) while this is set:
> > ```
> >  <PropertyGroup>
> >    <OutputType>Exe</OutputType>
> >    <TargetFramework>net8.0</TargetFramework>
> >    <ImplicitUsings>enable</ImplicitUsings>
> >    <Nullable>enable</Nullable>
> >    **<IsAspireHost>true</IsAspireHost>**
> >  </PropertyGroup>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you remove the IsAspireHost setting though it will see the reference/namespace and build successfully.
> > Why is this important?
> > 
> > * If you are using EF core and need to migrate the db created in the AppHost before the services come up you can't reference it.
> > * You can't create any supporting libraries for Aspire to handle your internal processes
> 
> Btw I on the other hand don't seem to can resolve project references. All `Microsoft.NET.Sdk.Web` projects are red to me on v1 with latest tooling VS 17.10 and 8.0.200
> 
> The `Microsoft.NET.Sdk` projects are fine
> 
> Got ideas?

You may have a similar issue I just fixed. I updated the Aspire workload but I also needed to update the nuget packages too. Once I did the Projects namespace fixed itself.

**Comment by murh-lego** [comment]
> > > In 8.0.0-preview.3.24105.21 and VS Version 17.9.0 when I add a link to a project library for supporting services it is ignored by the build system.
> > > Steps to Reproduce:
> > > 
> > > 1. Create an Aspire app using the standard VS template
> > > 2. Create a support library project and add any class to it
> > > 3. Reference that project in the Aspire apphost
> > > 4. Attempt to use the namespace for that support class from the external library
> > > 
> > > The build process refuses to accept those references when building (saying the namespace does not exist) while this is set:
> > > ```
> > >  <PropertyGroup>
> > >    <OutputType>Exe</OutputType>
> > >    <TargetFramework>net8.0</TargetFramework>
> > >    <ImplicitUsings>enable</ImplicitUsings>
> > >    <Nullable>enable</Nullable>
> > >    **<IsAspireHost>true</IsAspireHost>**
> > >  </PropertyGroup>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > If you remove the IsAspireHost setting though it will see the reference/namespace and build successfully.
> > > Why is this important?
> > > 
> > > * If you are using EF core and need to migrate the db created in the AppHost before the services come up you can't reference it.
> > > * You can't create any supporting libraries for Aspire to handle your internal processes
> > 
> > 
> > Btw I on the other hand don't seem to can resolve project references. All `Microsoft.NET.Sdk.Web` projects are red to me on v1 with latest tooling VS 17.10 and 8.0.200
> > The `Microsoft.NET.Sdk` projects are fine
> > Got ideas?
> 
> You may have a similar issue I just fixed. I updated the Aspire workload but I also needed to update the nuget packages too. Once I did the Projects namespace fixed itself.

I don't want to because there are other issues linked with v3 for me.
https://github.com/dotnet/aspire/issues/2226
My services take up ports on their own. Aspire creates proxy with same port and that won't work.

I'd like to use v1 the same way as before.
However, new version of VS / Workload / Dotnet break v1 for me (Sdk.Web projects can't be found)

**Comment by MiTschMR** [comment]
This may be the same issue as in https://github.com/dotnet/aspire/issues/2229 .

**Comment by eerhardt** [comment]
> Can you try appending IsAspireProjectResource="false" ?
> Is this going to be a long term way of dealing with that, or is this a short term fix?

That is the current plan of record for enabling this. There are a lot of discussion here at:

* https://github.com/dotnet/aspire/issues/1074
* https://github.com/dotnet/aspire/issues/1563

We tried a few different approaches, but the current behavior is what we landed on. MSBuild and NuGet have some issues that we can't tell if you are referencing a library or an exe early enough to be able to default this value at the right time.

Closing as I believe this issue is now addressed with setting `IsAspireProjectResource="false"`.
